### PR TITLE
132 create the service reaaaaaaa

### LINKED ISSUE #132: Create the Service REAAAAAAA
**Short description:**
You have to create a service that have only one reaction that enable or disable an ARea.

**Describe the solution you'd like**
1 reaction : Change ARea status
It takes two parameters : The ARea to modify and the status to set.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to add the service Reaaaaaaa

Fixes #132 

## Added features

* The Reaaaaaaa service is now available !
* A new reaction have been implemented (Change the status of an Area)
* Two getter url have been added for this reaction

** BE CAREFUL, two AREA cannot have the same name anymore after this PR **

* A sync button is now available on the Flutter interface to proceed a Force Re Sync of the object
* The Reaaaaaaa logo is available

## Updated features

* The Trigger map have been moved to avoid Dependency conflict

## Fixed feature

* A major bug of the Server is now fixed ! (Double tap on the Update button to successfully updating an AREA)

### CODE REVIEW

**Comment by TomDUVAL-MAHE** [comment]
** BE CAREFUL, two AREA cannot have the same name anymore after this PR **

WHY ???

**Review by tbellicha — DISMISSED** [review]

File: server/server_app.js [hunk 1]
```diff
@@ -574,6 +579,23 @@ app.post(
...
 )
 
+/**
+ * Cross all the User area and check the Name validity
+ * @param {*} userId the user Id
+ * @param {*} newName the wanted name
+ * @returns True if the new name is valid, false otherwise
+ */
+async function checkAreaNameAlreadyExistForGivenUser (userId, newName) {
+  let integrity = true
```
Comment (ZQUEMA) [inline server/server_app.js]: You check the validity, not the integrity

File: server/services/reaaaaaaa/reactions/change_area_status.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ).value
+    let newStatus = reactionParameters.find(
+      parameter => parameter.Parameter.name == 'newStatus'
+    ).value
+    changingAreaId = replaceDynamicParameters(changingAreaId, dynamicParameters)
+    newStatus = replaceDynamicParameters(newStatus, dynamicParameters)
+    const changingArea = await database.prisma.UsersHasActionsReactions.update({
+      where: { id: changingAreaId },
+      data: {
+        isEnable: newStatus == 'True' ? true : false
```
Comment (ZQUEMA) [inline server/services/reaaaaaaa/reactions/change_area_status.js]: I think that the ternary is not necessary here. The comparaison should be sufficient

File: server/services/timetime/init.js [hunk 3]
```diff
@@ -18,15 +18,15 @@ global.TimeTimeJobList = []
 /**
  * Map containing all the Trigger init function
  */
-const TriggerInitMap = {
+global.TriggerInitMap = {
```
Comment (ZQUEMA) [inline server/services/timetime/init.js:21]: Why do you define it as a global variable?

**Review by ZQUEMA — APPROVED** [review]
Few changes needed and answer to questions

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
Miss clicked last time

**Review by ZQUEMA — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]
NTR

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Pull request merged**